sdk/docs/pages/typescript/transaction-building/sponsored-transactions.mdx

```diff
@@ -4,17 +4,31 @@ The transaction block builder can support sponsored transaction blocks by using
 `onlyTransactionKind` flag when building the transaction block.
 
 ```tsx
-const txb = new TransactionBlock();
 
-// ... add some transactions...
+    const txb = new TransactionBlock();
 
-const kindBytes = await txb.build({ provider, onlyTransactionKind: true });
+    // ... add some commands...
 
-// construct a sponsored transaction from the kind bytes
-const sponsoredTxb = TransactionBlock.fromKind(kindBytes);
+    const kindBytes = await txb.build({client: client, onlyTransactionKind: true});
 
-// you can now set the sponsored transaction data that is required
-sponsoredTxb.setSender(sender);
-sponsoredTxb.setGasOwner(sponsor);
-sponsoredTxb.setGasPayment(sponsorCoins);
+    // construct a sponsored transaction from the kind bytes
+
+    const tempTx = TransactionBlock.fromKind(kindBytes);
+    tempTx.setSender(userAddress);
+    tempTx.setGasOwner(sponsorAddress);
+
+    //Collect signature from Sponsor
+    let sponsoredTx =  await sponsorKeypair.signTransactionBlock(await tempTx.build({client: client}));
+
+        //Collect signature from Sender
```

Comment:
indentation here looks off, you should be able to format this file with prettier